lms/services/lti_user.py

```diff
@@ -19,20 +19,24 @@ def __init__(
         self._lti_roles_service = lti_role_service
         self._application_instance_service = application_instance_service
 
-    def from_auth_params(self, application_instance, lti_core_schema) -> LTIUser:
-        """Create an LTIUser from a LTIV11CoreSchema like dict."""
+    def from_lti_params(self, application_instance, lti_params) -> LTIUser:
+        """Create an LTIUser from a LTIParams."""
 
         return self.deserialize(
-            user_id=lti_core_schema["user_id"],
+            user_id=lti_params["user_id"],
             application_instance_id=application_instance.id,
-            roles=lti_core_schema["roles"],
-            tool_consumer_instance_guid=lti_core_schema["tool_consumer_instance_guid"],
+            roles=lti_params["roles"],
+            tool_consumer_instance_guid=lti_params["tool_consumer_instance_guid"],
             display_name=display_name(
-                lti_core_schema["lis_person_name_given"],
-                lti_core_schema["lis_person_name_family"],
-                lti_core_schema["lis_person_name_full"],
+                lti_params.get("lis_person_name_given", ""),
```

Comment:
The previous values come from an marshmallow dict, "" was the default value for these.

Using the same behavior and adding an explicit test.

Reply:
Big yes to not passing around plain dicts from marshmallow